packages/react-aria-components/docs/Tree.mdx

```diff
@@ -237,19 +237,530 @@ let items = [
+
+```tsx example
+<PokemonEvolutionTree selectionMode="multiple" selectionBehavior="replace" />
+```
+
+## Item actions
+
+`Tree` supports item actions via the `onAction` prop, which is useful for functionality such as navigation. In the default `"toggle"` selection behavior, when nothing is selected, clicking or tapping the item triggers the item action.
+When at least one item is selected, the tree is in selection mode, and clicking or tapping an item toggles the selection. Actions may also be triggered via the <Keyboard>Enter</Keyboard> key, and selection using the <Keyboard>Space</Keyboard> key.
+
+This behavior is slightly different in the `"replace"` selection behavior, where single clicking selects the item and actions are performed via double click. On touch devices, the action becomes the primary tap interaction,
+and a long press enters into selection mode, which temporarily swaps the selection behavior to `"toggle"` to perform selection (you may wish to display checkboxes when this happens). Deselecting all items exits selection mode
+and reverts the selection behavior back to `"replace"`. Keyboard behaviors are unaffected.
+
+```tsx example
+<div style={{display: 'flex', flexWrap: 'wrap', gap: '24px'}}>
+  <PokemonEvolutionTree
+    aria-label="Pokemon tree with item actions and toggle selection behavior"
+    ///- begin highlight -///
+    onAction={key => alert(`Opening item ${key}...`)}
+    ///- end highlight -///
+    selectionMode="multiple" />
+  <PokemonEvolutionTree
+    aria-label="Pokemon tree with item actions and replace selection behavior"
+    ///- begin highlight -///
+    onAction={key => alert(`Opening item ${key}...`)}
+    selectionBehavior="replace"
+    ///- end highlight -///
+    selectionMode="multiple" />
+</div>
+```
+
+Items may also have an action specified by directly applying `onAction` on the `TreeItem` itself. This may be especially convenient in static collections. If `onAction` is also provided to the `Tree`, both the trees's and the item's `onAction` are called.
+
+```tsx example
+<Tree aria-label="Tree with onAction applied on the items directly" style={{height: '300px'}} defaultExpandedKeys={['bulbasaur', 'ivysaur']}>
+  {/*- begin highlight -*/}
+  <TreeItem id="bulbasaur" textValue="Bulbasaur" onAction={() => alert(`Opening Bulbasaur...`)}>
```

Comment:
You can remove the ids from these to keep the example shorter.

Wonder if we can also infer textValue from the children of TreeItemContent if it is plain text like we do in GridList. Or you could use `MyTreeItem` here and do it there?

Reply:
the id is useful if i want to use a legible and obvious defaultExpandedKeys